wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitJsonRpcMessenger.java

```diff
@@ -67,6 +86,26 @@ public void onEvent(GitCheckoutEvent event) {
                     .sendAndSkipResult());
   }
 
+  private void handleIndexChangedEvent(GitEvent event) {
+    Status status = newDto(Status.class);
+    if (event instanceof GitCommitEvent) {
+      status = ((GitCommitEvent) event).getStatus();
+    } else if (event instanceof GitResetEvent) {
+      status = ((GitResetEvent) event).getStatus();
+    } else if (event instanceof GitRepositoryInitializedEvent) {
+      status = ((GitRepositoryInitializedEvent) event).getStatus();
+    }
+
+    for (String endpointId : endpointIds) {
+      transmitter
+          .newRequest()
+          .endpointId(endpointId)
+          .methodName("event/git-index")
```

Comment:
May be we could use here something like "event/git/indexStatusChanged" or "event/git/indexChanged" similar to previously used "event/git/subscribe" just to keep homogeneous method names here?

Reply:
I'm actually wondering if we should add one more level to the method naming scope, I mean we could use `event/git/index/subscribe` and then `event/git/index/statusChanged` or `event/git/index/changed`, what do you think?

Reply:
This chanel is already used in `GitIndexChangedDetector` so I am reusing it here

Reply:
Well, I'm just assuming that we probably could change it if it would fit the logic better :-)